daml-assistant/daml-helper/src/DA/Daml/Helper/Start.hs

```diff
@@ -225,3 +260,68 @@ runStart
             case mbJsonApiPort of
                 Nothing -> f sandboxPh
                 Just jsonApiPort -> withJsonApi sandboxPort jsonApiPort args f
+
+platformVersionEnvVar :: String
+platformVersionEnvVar = "DAML_PLATFORM_VERSION"
+
+-- | Returns the platform version determined as follows:
+--
+-- 1. If DAML_PLATFORM_VERSION is set return that.
+-- 2. If DAML_PROJECT is set and non-empty and `daml.yaml`
+--    has a `platform-version` field return that.
+-- 3. If `DAML_SDK_VERSION` is set return that.
+-- 4. Else we are invoked outside of the assistant and we throw an exception.
+getPlatformVersion :: IO String
+getPlatformVersion = do
+  mbPlatformVersion <- lookupEnv platformVersionEnvVar
+  case mbPlatformVersion of
+    Just platformVersion -> pure platformVersion
+    Nothing -> do
+      mbProjPath <- getProjectPath
+      case mbProjPath of
+        Just projPath -> do
+          project <- readProjectConfig (ProjectPath projPath)
+          case queryProjectConfig ["platform-version"] project of
+            Left err -> throwIO err
+            Right (Just ver) -> pure ver
+            Right Nothing -> getSdkVersion
+        Nothing -> getSdkVersion
+
+-- Convenience-wrapper around `withPlatformProcess` for commands that call the SDK
+-- JAR `daml-sdk.jar`.
+withPlatformJar
+    :: [String]
+    -- ^ Commands passed to the assistant and the platform JAR.
+    -> FilePath
+    -- ^ File name of the logback config.
+    -> (Process () () () -> IO a)
+    -> IO a
+withPlatformJar args logbackConf f = do
+    logbackArg <- getLogbackArg (damlSdkJarFolder </> logbackConf)
```

Comment:
Does this use the logback config from the current SDK to run the platform version JAR, or, does this use the current SDK config to determine the location of the logback config in the platform version's SDK? Both of these seem bad, but the latter seems worse IMO.

I don't know a way around this other than re-invoking daml-assistant for the necessary component with a modified DAML_SDK_VERSION, in the cases where the platform version /= SDK version. I.e. we shouldn't be invoking jars across SDK versions directly.

If the former, how often do logback configs change? Does running an SDK component with a different logback config affect our ability to hit compatibility targets? My guess is no...

We should try really hard to avoid getting into a situation that makes it difficult to change the SDK / platform in the future.

Reply:
The logback arg is only used in the second argument to `withPlatformProcess` which is only used if we are running a component from the current SDK. If we run a component from the platform version we go via the assistant and don’t pass the logback config.

At least that’s what I intended to do, am I missing something?

Reply:
Ah, I'm silly. Sorry, of course, you're doing the right thing, but I got worried prematurely and then got distracted and didn't read all the way down to `withPlatformProcess`. Oops, sorry.